vendor_pricelist_disable/tests/test_purchase_pricelist_disable.py

```diff
@@ -0,0 +1,192 @@
+
+
+@tagged("post_install", "-at_install")
+class TestPurchasePricelistDisable(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.product = cls.env["product.product"].create(
+            {
+                "name": "Test Product",
+                "type": "product",
+            }
+        )
+
+        cls.partner1 = cls.env["res.partner"].create(
+            {
+                "name": "Vendor with auto pricelist",
+            }
+        )
+
+        cls.partner2 = cls.env["res.partner"].create(
+            {
+                "name": "Vendor without auto pricelist",
+            }
+        )
+
+        cls.company2 = cls.env["res.company"].create(
+            {
+                "name": "Test Company",
+            }
+        )
+
+    def test_01_pricelist_enabled(self):
+        """Test supplier pricelist creation when enabled"""
+        self.env.company.purchase_pricelist_disable_autocreate = True
+
+        po = self.env["purchase.order"].create(
+            {
+                "partner_id": self.partner1.id,
+            }
+        )
+        self.env["purchase.order.line"].create(
+            {
+                "order_id": po.id,
+                "product_id": self.product.id,
+                "product_qty": 1.0,
+                "price_unit": 100.0,
+                "name": "Test Line",
+                "product_uom": self.product.uom_id.id,
+            }
+        )
+
+        seller_count_before = len(self.product.seller_ids)
+
+        po.button_confirm()
+
+        seller_count_after = len(self.product.seller_ids)
+        self.assertEqual(seller_count_after, seller_count_before + 1)
+        self.assertIn(self.partner1, self.product.seller_ids.mapped("partner_id"))
+
+    def test_02_pricelist_disabled(self):
+        """Test supplier pricelist creation when disabled"""
+        self.env.company.purchase_pricelist_disable_autocreate = False
+
+        po = self.env["purchase.order"].create(
+            {
+                "partner_id": self.partner2.id,
+            }
+        )
+        self.env["purchase.order.line"].create(
+            {
+                "order_id": po.id,
+                "product_id": self.product.id,
+                "product_qty": 1.0,
+                "price_unit": 100.0,
+                "name": "Test Line",
+                "product_uom": self.product.uom_id.id,
+            }
+        )
+
+        seller_count_before = len(self.product.seller_ids)
+
+        po.button_confirm()
+
+        seller_count_after = len(self.product.seller_ids)
+        self.assertEqual(seller_count_after, seller_count_before)
+        self.assertNotIn(self.partner2, self.product.seller_ids.mapped("partner_id"))
+
+    def test_03_multiple_products_pricelist_enabled(self):
+        """Test multiple supplier pricelists creation when enabled"""
+        self.env.company.purchase_pricelist_disable_autocreate = True
+
+        product2 = self.env["product.product"].create(
```

Comment:
You can create the product and the order in the setUpClass method, so you don't duplicate code in test_03_multiple_products_pricelist_enabled and test_04_multiple_products_pricelist_disabled